isthmus/src/test/java/io/substrait/isthmus/PlanTestBase.java

```diff
@@ -51,4 +58,35 @@ protected void assertPlanRoundrip(Plan plan) throws IOException, SqlParseExcepti
         new PlanProtoConverter().toProto(new ProtoPlanConverter().from(protoPlan1));
     assertEquals(protoPlan1, protoPlan2);
   }
+
+  protected List<RelNode> assertSqlSubstraitRelRoundTrip(String query) throws Exception {
+    // sql <--> substrait round trip test.
+    // Assert (sql -> substrait) and (sql -> substrait -> calcite rel -> substrait) are same.
+    // Return list of sql -> substrait rel -> Calcite rel.
+    String[] values = asString("tpch/schema.sql").split(";");
+    var creates = Arrays.stream(values).filter(t -> !t.trim().isBlank()).toList();
+    List<RelNode> relNodeList = new ArrayList<>();
+
+    // 1. sql -> substrait rel
+    SqlToSubstrait s = new SqlToSubstrait();
+    for (RelRoot relRoot : s.sqlToRelNode(query, creates)) {
+      Rel pojoRel = SubstraitRelVisitor.convert(relRoot, EXTENSION_COLLECTION);
+
+      // 2. substrait rel -> Calcite Rel
+      RelNode relnodeRoot = new SubstraitToSql().substraitRelToCalciteRel(pojoRel, creates);
+
+      relNodeList.add(relnodeRoot);
+
+      // 3. Calcite Rel -> substrait rel
+      Rel pojoRel2 =
+          SubstraitRelVisitor.convert(
+              RelRoot.of(relnodeRoot, SqlKind.SELECT), EXTENSION_COLLECTION);
+
+      Assertions.assertEquals(pojoRel, pojoRel2);
+
+      // 4. Calcite Rel -> sql
+      String convertedSql = SubstraitToSql.toSql(relnodeRoot);
```

Comment:
Is this conversion just to see if you get an exception or not? Would it be possible to go back from sql to proto rel or pojo rel and compare it's the same?

Reply:
Previously, this conversion is to return a sql and assert Sql contains certain substring. Later, I changed to return RelNode, and verify RelNode tree contains certain expression.  

I tried to go back from sql to pojo rel and do the compare. The problem is `toSql` may loose some information. Consider the following query:
```
select p_size  from part where p_partkey > cast(100 as bigint)
``` 

In the  first round sql -> RelNode -> substrait rel ->  Calcite rel,  the condition is essentially represented as
`p_partkey > 100:bigint`.  However, from calcite rel -> sql, the condition becomes `p_partkey > 100`.  If we convert such sql back to substrait rel, the expression would contains a `cast`. Although they should return a same resultset, the comparison of pojo rel would fail.

In another words,  `toSql()` produces semantic equivalent yet different query.  Probably,  I should remove this `toSql` step.